File: src/main/scala/com/gaastats/util/ResourceHelper.java
package com.gaastats.util;

import android.app.Activity;
import android.app.Application;
import android.app.FragmentManager;
import android.content.Context;
import android.view.View;

import com.google.inject.Inject;
import com.google.inject.Provider;
import com.google.inject.Singleton;

@Singleton
public class ResourceHelper {
	@Inject
	private Provider<Application> application;
	private Activity activity;
	
	public void setActivity(Activity activity) {
		this.activity = activity;
	}

	public View findViewById(int viewID) {
		return activity.findViewById(viewID);
	}
	
	public Context getContext() {
		return application.get();
	}
	
	public FragmentManager getFragmentManager() {
		return activity.getFragmentManager();
	}
}

File: src/main/scala/com/gaastats/util/ResourceHelper.scala
package com.gaastats.util

import android.app.{Activity, Application, FragmentManager}
import android.view.View


/**
 * Created by David on 2/16/2015.
 */
object ResourceHelper {
  implicit var application: Application = null

  def setApplication(application: Application) {
    this.application = application
  }
}
